core/trino-main/src/main/java/io/trino/metadata/MetadataUtil.java

```diff
@@ -202,6 +203,20 @@ public static boolean tableExists(Metadata metadata, Session session, String tab
         return metadata.getTableHandle(session, name).isPresent();
     }
 
+    public static void checkRoleExists(Session session, Node node, Metadata metadata, String role, String catalog)
+    {
+        if (!metadata.roleExists(session, role, catalog)) {
+            throw semanticException(ROLE_NOT_FOUND, node, "Role '%s' does not exist in catalog '%s'", role, catalog);
+        }
+    }
+
+    public static void checkRoleExists(Session session, Node node, Metadata metadata, TrinoPrincipal principal, String catalog)
```

Comment:
nit: Please move this method one method up, so they are ordered by the usage from high level detail to low level.